### ISSUE TITLE
1.7.8.5 - The product does not appear in the shopping cart.

### ISSUE THREAD

**Issue by Akmenelis** [PrestaShop/PrestaShop#28213]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Hello,

After clicking buy, the product does not appear in the shopping cart.
![Ekrano nuotrauka 2022-04-03 205258](https://user-images.githubusercontent.com/103214415/162629546-30987125-8a54-4459-85fb-2f69bf805f00.png)

I use Standart installed theme and shopping cart.
PHP version 7.4.14
When MySQL opens, it shows the items in the cart.

Now run debug mode on shopping cart:

https://www.xxxxx.xxx/en/module/ps_shoppingcart/ajax?action=add-to-cart

And get this:

```
_Whoops, looks like something went wrong.

(1/1) ContextErrorException

Notice: Trying to access array offset on value of type null

in 0de54a7df2ce9f325675562cb8b6164899785ff9_2.module.psshoppingcartmodal.tpl.php line 40

at content_6251c7b6e6fdf8_09239384()in smarty_template_resource_base.php line 123

at Smarty_Template_Resource_Base->getRenderedTemplateCode()in smarty_template_compiled.php line 114

at Smarty_Template_Compiled->render()in smarty_internal_template.php line 216

at Smarty_Internal_Template->render()in smarty_internal_templatebase.php line 232

at Smarty_Internal_TemplateBase->_execute()in smarty_internal_templatebase.php line 116

at Smarty_Internal_TemplateBase->fetch()in SmartyDevTemplate.php line 40

at SmartyDevTemplateCore->fetch()in Module.php line 2490

at ModuleCore->fetch()in ps_shoppingcart.php line 148

at Ps_Shoppingcart->renderModal()in ajax.php line 43

at Ps_ShoppingcartAjaxModuleFrontController->initContent()in Controller.php line 306

at ControllerCore->run()in Dispatcher.php line 518

at DispatcherCore->dispatch()in index.php line 28_
```

 

Any ideas?

 

I tried disabling geolocation, deleting the var folder, disabling the friendly url, I tried to disable / uninstall / install cart module, clear the cache, change browsers. Does not work.

### Expected behavior

I hope at least advice on where and what code to adjust, as I am a new user with some experience with php.

### Steps to reproduce

1. Go as customer to buy.
2. Click buy.
3. Shows empty cart with message: Product successfully added to your shopping cart

### PrestaShop version(s) where the bug happened

1.7.8.5

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

ps_shoppingcart

**Comment by hibatallahAouadni** [comment]
Hello @Akmenelis 

I didn't manage to reproduce the issue with PS1785, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/162706416-9e41b4c8-6d7c-4c50-97f1-090c2fab9a01.png)

For the link, I get this:

```
{
  "preview":
    "\n<!-- begin module:ps_shoppingcart\/ps_shoppingcart.tpl -->\n<!-- begin \/projects\/prestashop_1.7.8.5\/themes\/classic\/modules\/ps_shoppingcart\/ps_shoppingcart.tpl --><div id=\"_desktop_cart\">\n  <div class=\"blockcart cart-preview active\" data-refresh-url=\"\/\/prestashop\/prestashop_1.7.8.5\/en\/module\/ps_shoppingcart\/ajax\">\n    <div class=\"header\">\n              <a rel=\"nofollow\" aria-label=\"Shopping cart link containing 1 product(s)\" href=\"\/\/prestashop\/prestashop_1.7.8.5\/en\/cart?action=show\">\n              <i class=\"material-icons shopping-cart\" aria-hidden=\"true\">shopping_cart<\/i>\n        <span class=\"hidden-sm-down\">Cart<\/span>\n        <span class=\"cart-products-count\">(1)<\/span>\n              <\/a>\n          <\/div>\n  <\/div>\n<\/div>\n<!-- end \/projects\/prestashop_1.7.8.5\/themes\/classic\/modules\/ps_shoppingcart\/ps_shoppingcart.tpl -->\n<!-- end module:ps_shoppingcart\/ps_shoppingcart.tpl -->\n","modal":"\n<!-- begin module:ps_shoppingcart\/modal.tpl -->\n<!-- begin \/projects\/prestashop_1.7.8.5\/themes\/classic\/modules\/ps_shoppingcart\/modal.tpl --><div id=\"blockcart-modal\" class=\"modal fade\" tabindex=\"-1\" role=\"dialog\" aria-labelledby=\"myModalLabel\" aria-hidden=\"true\">\n  <div class=\"modal-dialog\" role=\"document\">\n    <div class=\"modal-content\">\n      <div class=\"modal-header\">\n        <button type=\"button\" class=\"close\" data-dismiss=\"modal\" aria-label=\"Close\">\n          <span aria-hidden=\"true\"><i class=\"material-icons\">close<\/i><\/span>\n        <\/button>\n        <h4 class=\"modal-title h6 text-sm-center\" id=\"myModalLabel\"><i class=\"material-icons rtl-no-flip\">&#xE876;<\/i>Product successfully added to your shopping cart<\/h4>\n      <\/div>\n      <div class=\"modal-body\">\n        <div class=\"row\">\n          <div class=\"col-md-5 divide-right\">\n            <div class=\"row\">\n              <div class=\"col-md-6\">\n                                  <img\n                    src=\"http:\/\/prestashop\/prestashop_1.7.8.5\/img\/p\/en-default-medium_default.jpg\"\n                    loading=\"lazy\"\n                    class=\"product-image\"\n                  \/>\n                              <\/div>\n              <div class=\"col-md-6\">\n                <h6 class=\"h6 product-name\"><\/h6>\n                <p class=\"product-price\"><\/p>\n                \n                                <span class=\"product-quantity\">Quantity:&nbsp;<strong><\/strong><\/span>\n              <\/div>\n            <\/div>\n          <\/div>\n          <div class=\"col-md-7\">\n            <div class=\"cart-content\">\n                              <p class=\"cart-products-count\">There is 1 item in your cart.<\/p>\n                            <p><span class=\"label\">Subtotal:<\/span>&nbsp;<span class=\"subtotal value\">\u20ac24.00<\/span><\/p>\n                              <p><span>Shipping:<\/span>&nbsp;<span class=\"shipping value\">Free <\/span><\/p>\n              \n                              <p class=\"product-total\"><span class=\"label\">Total&nbsp;(tax incl.)<\/span>&nbsp;<span class=\"value\">\u20ac24.00<\/span><\/p>\n              \n                            \n              <div class=\"cart-content-btn\">\n                <button type=\"button\" class=\"btn btn-secondary\" data-dismiss=\"modal\">Continue shopping<\/button>\n                <a href=\"\/\/prestashop\/prestashop_1.7.8.5\/en\/cart?action=show\" class=\"btn btn-primary\"><i class=\"material-icons rtl-no-flip\">&#xE876;<\/i>Proceed to checkout<\/a>\n              <\/div>\n            <\/div>\n          <\/div>\n        <\/div>\n      <\/div>\n      \n    <\/div>\n  <\/div>\n<\/div>\n<!-- end \/projects\/prestashop_1.7.8.5\/themes\/classic\/modules\/ps_shoppingcart\/modal.tpl -->\n<!-- end module:ps_shoppingcart\/modal.tpl -->\n"
}
```

Could you please Reset your module and check if you still encounter the same issue.
Waiting for your feedback.

Thanks!

**Comment by Akmenelis** [comment]
When you click buy - your cart is completely empty. Not only does it not display product photos, but it also does not display product name and quantity and price. When you try to buy another product, the cart is still completely empty.
![InkedEkrano nuotrauka 2022-04-03 205258_LI](https://user-images.githubusercontent.com/103214415/162792711-d8336610-ee8a-4fde-bc5b-e60daf16b3e5.jpg)

When I open mysql, I see those products in the cart. As I understand it, information from mysql is not displayed.

https://user-images.githubusercontent.com/103214415/162797305-ecae301e-a850-44bc-bb48-88ed5b76384d.mp4

**Comment by marwachelly** [comment]
Hello @Akmenelis,

It seems that you have a cache problem. Could you pleae try to clean your cache browser or use a new Incognito Window ? 

Could you please check and feedback?

Thanks,

**Comment by Akmenelis** [comment]
I tried clering my browser cache, i tried to open a webpage from other browsers and devices. I also deleted var folder on server that contains cache. The trouble does not go away.

**Comment by marwachelly** [comment]
Hello @Akmenelis ,

Despite our efforts, we were not able to reproduce your issue with the provided information.

https://user-images.githubusercontent.com/89604350/162977013-f8f1e04e-6834-4d7f-ab7b-198011886a30.mp4



It seems that your issue is not a PrestaShop core bug but most likely a module problem.
If you have questions about using PrestaShop or third-party modules, or if you need help with your shop, please consider one of our [support plans](https://www.prestashop.com/support).
Alternatively, you can also ask for help in the [community forums](https://www.prestashop.com/forums/) or in the [public Slack channel](https://prestashop.slack.com/).

We will close this issue.

Thanks,
